app/controllers/instructeurs/procedures_controller.rb

```diff
@@ -324,8 +325,48 @@ def apercu
       DossierPreloader.load_one(@dossier)
     end
 
+    def history
+      @procedure = procedure
+      all_published_revisions = @procedure.revisions.where.not(published_at: nil).order(published_at: :asc)
+
+      @comparison_pairs = []
+
+      if all_published_revisions.size >= 2
+        1.upto(all_published_revisions.size - 1) do |i|
+          previous = all_published_revisions[i - 1]
+          current = all_published_revisions[i]
+
+          @comparison_pairs << [current, previous]
+        end
+
+        @comparison_pairs.reverse!
+      end
+
+      @instructeur_procedure = InstructeursProcedure.find_by(
+        instructeur: current_instructeur,
+        procedure: @procedure
+      )
```

Comment:
je pense que la aussi on peut faire mieux pour eviter le cas du @instructeur_procedure nil

```suggestion
      @instructeur_procedure = InstructeursProcedure
        .create_with(last_revision_seen_id: procedure.published_revision_id)
        .find_or_create_by(instructeur: current_instructeur, procedure: @procedure)
```